File: src/app/movie.ts
import { Review } from "./reviews";

export interface Movie {
  id?: number | null;
  title: string;
  genre: string;
  releaseDate: string;
  director: string;
  reviews?: Review[];  // Optional, list of reviews associated with this movie
}

export interface NewMovie {

  title: string;
  genre: string;
  releaseDate: string;
  director: string;
}

File: MovieManagementAPI/Models/Movie.cs
namespace MovieManagementAPI.Models
{
    public class Movie
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty; // Default value
        public string Genre { get; set; } = string.Empty;
        public DateTime ReleaseDate { get; set; }
        public string Director { get; set; } = string.Empty;
        public List<Review>? Reviews { get; set; }
    }

}
